sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_resource_configuration.py

```diff
@@ -50,11 +50,15 @@ def as_dict(self) -> Dict[str, Any]:
 
     @classmethod
     def _to_dict(cls, obj: Any) -> Any:
+        # pylint: disable=protected-access
         if isinstance(obj, dict):
+            need_key_mapping = getattr(obj, "_NEED_KEY_MAPPING", False)
```

Comment:
Add some comment for this? Can we do this by checking existence of `_KEY_MAPPING`?

Reply:
Sure, and I think checking existence shall work, let me update this.